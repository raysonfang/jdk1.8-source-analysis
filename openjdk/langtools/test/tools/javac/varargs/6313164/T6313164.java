/*
 * @test /nodynamiccopyright/
 * @bug     6313164
 * @author mcimadamore
 * @summary  javac generates code that fails byte code verification for the varargs feature
 * @compile/fail/ref=T6313164.out -XDrawDiagnostics T6313164.java
 */
import p1.*;

class T6313164 {
    { B b = new B();
      b.foo1(new B(), new B()); //error - A not accesible
      b.foo2(new B(), new B()); //ok - A not accessible, but foo2(Object...) applicable
      b.foo3(null, null); //error - A (inferred) not accesible
      b.foo4(null, null); //error - A (inferred in 15.12.2.8 - no resolution backtrack) not accesible
      b.foo4(new B(), new C()); //ok - A (inferred in 15.12.2.7) not accessible, but foo4(Object...) applicable
    }
}
