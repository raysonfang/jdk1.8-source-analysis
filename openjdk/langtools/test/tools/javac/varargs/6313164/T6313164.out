T6313164.java:12:8: compiler.err.cant.apply.symbol: kindname.method, foo1, p1.A[], p1.B,p1.B, kindname.class, p1.B, (compiler.misc.inaccessible.varargs.type: p1.A, kindname.class, T6313164)
T6313164.java:14:13: compiler.err.prob.found.req: (compiler.misc.inaccessible.varargs.type: p1.A, kindname.class, T6313164)
T6313164.java:15:13: compiler.err.prob.found.req: (compiler.misc.inaccessible.varargs.type: p1.A, kindname.class, T6313164)
- compiler.note.unchecked.filename: B.java
- compiler.note.unchecked.recompile
3 errors
